x-pack/filebeat/module/zeek/dns/config/dns.yml

```diff
@@ -6,39 +6,132 @@ paths:
 exclude_files: [".gz$"]
 tags: {{.tags}}
 
-json.keys_under_root: false
-
 processors:
-  - drop_fields:
-      fields: ["json.Z","json.auth","json.addl"]
   - rename:
       fields:
-        - from: "json"
-          to: "zeek.dns"
-
-        - from: "zeek.dns.id.orig_h"
-          to: "source.address"
+        - {from: message, to: log.original}
+  - decode_json_fields:
+      fields: [log.original]
+      target: zeek.dns
+  - script:
+      lang: javascript
+      id: zeek_dns_flags
+      source: >
+        function addDnsHeaderFlags(evt) {
+            var flag = evt.Get("zeek.dns.AA");
+            if (flag === true) {
+                evt.AppendTo("dns.header_flags", "AA");
+            }
+            flag = evt.Get("zeek.dns.TC");
+            if (flag === true) {
+                evt.AppendTo("dns.header_flags", "TC");
+            }
+            flag = evt.Get("zeek.dns.RD");
+            if (flag === true) {
+                evt.AppendTo("dns.header_flags", "RD");
+            }
+            flag = evt.Get("zeek.dns.RA");
+            if (flag === true) {
+                evt.AppendTo("dns.header_flags", "RA");
+            }
+            flag = evt.Get("zeek.dns.Z");
+            if (flag === true) {
+                evt.AppendTo("dns.header_flags", "Z");
+            }
+        }
 
-        - from: "zeek.dns.id.orig_p"
-          to: "source.port"
+        function addDnsQuestionClass(evt) {
+            var qclass = evt.Get("zeek.dns.qclass");
+            if (!qclass) {
+                return;
+            }
+            switch (qclass) {
+                case 1:
+                    qclass = "IN";
+                    break;
+                case 3:
+                    qclass = "CH";
+                    break;
+                case 4:
+                    qclass = "HS";
+                    break;
+                case 254:
+                    qclass = "NONE";
+                    break;
+                case 255:
+                    qclass = "ANY";
+                    break;
+            }
+            evt.Put("dns.question.class", qclass);
+        }
 
-        - from: "zeek.dns.id.resp_h"
-          to: "destination.address"
+        function addDnsAnswers(evt) {
+            var answers = evt.Get("zeek.dns.answers");
+            var ttls = evt.Get("zeek.dns.TTLs");
+            if (!answers || !ttls || answers.length != ttls.length) {
```

Comment:
I would hope that doesn't happen, but good check.

But if it does happen, I would suggest appending an error message to that effect to `error.message` instead of just returning.

Not a blocker, though. Can also be improved later.